### DISCUSSION TITLE
Sonoff L1 Lite LED Strip (Spider Z) no control of LEDS

### DISCUSSION THREAD

**Discussion by kyemacdonald** [arendst/Tasmota#12930] [Q&A] [answered]
Hi All,
I have successfully flashed a Sonoff Spider Z LED controller that came as part of a Sonoff L1 Lite LED Strip kit.  Flashing went fine and I can now access the tasmota interface on the device.

But i have no direct control of the LEDS.  I can turn the entire strip on or off, but that is it.  No colour, brightness or saturation though I have those control options in the webpage.  Whats more is it still responds to the IR remote.  

I have previously flashed Spider Z boards that come with the more expensive L1 kit and those are working perfectly.  

I have used the template {"NAME":"Sonoff L1 Lite","GPIO":[0,3200,0,3232,0,0,0,0,0,320,0,0,0,0],"FLAG":0,"BASE":70} taken from https://templates.blakadder.com/sonoff_L1_Lite.html that is the exact device I am trying to use.

Any advice, things to try etc would be appreciated.

**Comment by barbudor** [comment]
Was it working with original firmware ?
Did you kept a copy of the original firmware ?

Tasmota do not directly drives the LEDs but talk through serial to another MCU.
If the protocol across the serial link has changed, someone will have to reverse engineer it from the original firmware and update Tasmota.

**Reply by kyemacdonald** [reply]
I never even tried it with the stock firmware... or keep a copy.  I've got several of these in place already and didn't even think to.

**Reply by kyemacdonald** [reply]
In the terminal output with full logging I can see this:
13:36:07.422 MQT: stat/TV-LEDS/RESULT = {"POWER":"ON","Dimmer":100,"Color":"00FF15","HSBColor":"125,100,100","Channel":[0,100,8]}
13:36:07.449 SL1: Send AT+UPDATE="sequence":"1629466567969","switch":"on","light_type":1,"colorR":0,"colorG":255,"colorB":21,"bright":100,"mode":1,"sensitive":5,"speed":50
13:36:07.585 SL1: Rcvd AT+RESULT="sequence":"1629466567969"
13:36:07.586 SL1: Send AT+SEND=ok
13:36:07.838 CFG: Saved to flash at FA, Count 126, Bytes 4096

But that has no impact on the actual LED colour / settings.  Is there anything I can try to brute force those commands?

**Reply by barbudor** [reply]
So far this confirm that Tasmota can talk to the MCU as it receive response to the AT+UPDATE message
The values in the serial messages seem to be in line with the MQTT status
The MCU response doesn't seem to indicate that it doesn't understand the message, so for now it would rules out a change in the protocol.
Could it be a defective unit ?
Especially if you have more of them working fine with the same tasmota revision

**Reply by kyemacdonald** [reply]
Potentially faulty I guess.  I have another due to arrive early next week and will test it with stock firmware first.  My existing one was on 9.4.0 and I just downgraded the new one to 9.4.0 to see if it made any difference but no change.

**Reply by barbudor** [reply]
After testing with the original firmware, I would suggest to keep a copy of the original firmware when flashing with Tasmota
You use that to revert this one to the original firmware for testing

**Reply by kyemacdonald** [reply]
Ok there are now other reports of people with the same issue, so it is definitely not a one off faulty hardware.  I have linked the others to this thread

**Comment by kyemacdonald** [comment]
Here are photos of the board.  Sonoff must have changed something recently as there are now multiple reports.  I don't see any obvious differences between the new and older boards, other than some slight screenprinting differences in the lower right below the LED connection point.

![20210822_092712](https://user-images.githubusercontent.com/51263971/130337294-a1150365-1b0e-4393-bb21-e39a7b4eb4b9.jpg)
![20210822_092743](https://user-images.githubusercontent.com/51263971/130337295-8ccd6677-25f0-4565-80c7-fff998613a4a.jpg)
![20210822_092735](https://user-images.githubusercontent.com/51263971/130337296-e0295a8e-2e19-490d-9c0f-6ace4165815b.jpg)
![20210822_092839](https://user-images.githubusercontent.com/51263971/130337297-95be4754-c405-4e9c-a6a5-97cd2c344bf4.jpg)

**Reply by barbudor** [reply]
As the hardware (LEDs) are under control of the MCU, I wouldn't be bothered by a change in the hardware design if the protocol is still the same.
So the best would still be to spy the serial between the ESP and the MCU with the original ESP firmware.

**Comment by mediarath** [comment]
I have also already confirmed this misbehavior in the chat. I have the same problem. Sonoff seems to have changed something. I have flashed 3 with the 9.5 and they have this exact behavior. All new. I happen to have a slightly older one unflashed. This one can be operated normally after flashing. At first glance I can't see any difference on the board.
Also the command l1musicsync has no effect.

I have now made a downgrade to the 9.2. and now it runs again. So apparently the L1musicsync implementation in the newer Firmwares with the "newer" L1 is no longer compatible. My old L1 also runs on the latest firmware. Even the latest 9.5.0.7 does not fix this problem.

**Reply by kyemacdonald** [reply]
Confirmed that the 9.2 firmware works with the newer Spider Zs.

**Reply by barbudor** [reply]
Can you confim that v9.3.0/9.3.1 are already impacted by the problem ?
If not, which exact oldest version shows the problem ?

**Reply by kyemacdonald** [reply]
Just tested.
9.3.1 - Doesn't work
9.3.0 - Doesn't work
9.2.0 - Works

**Reply by barbudor** [reply]
Thanks.
This is helpful to narrow down which changes may have impacted

Another question  can you check the result of command `L1MusicSync` with a recent version ?
Mode should be 0. If not try after entering the command `L1MusicSync 0`

**Reply by kyemacdonald** [reply]
Version 9.2
22:53:16 CMD: L1MusicSync
22:53:16 SRC: WebConsole from 10.1.1.34
22:53:16 CMD: Group 0, Index 1, Command "L1MUSICSYNC", Data ""
22:53:16 MQT: stat/TV-LEDS/RESULT = {"Command":"Unknown"}

Version 9.5.0
22:59:54.473 CMD: l1musicsync
22:59:54.475 SRC: WebConsole from 10.1.1.34
22:59:54.477 CMD: Grp 0, Cmnd 'L1MUSICSYNC', Idx 1, Len 0, Data ''
22:59:54.485 MQT: stat/TV-LEDS/RESULT = {"L1MusicSync":{"Mode":"OFF","Sensitive":5,"Speed":50}}

23:01:02.430 CMD: l1musicsync 0
23:01:02.433 SRC: WebConsole from 10.1.1.34
23:01:02.435 CMD: Grp 0, Cmnd 'L1MUSICSYNC', Idx 1, Len 1, Data '0'
23:01:02.440 SL1: Send AT+UPDATE="sequence":"1629759662963","switch":"on","light_type":1,"colorR":76,"colorG":49,"colorB":77,"bright":100,"mode":1,"sensitive":5,"speed":50
23:01:02.523 MQT: stat/TV-LEDS/RESULT = {"L1MusicSync":{"Mode":"OFF","Sensitive":5,"Speed":50}}
23:01:02.577 SL1: Rcvd AT+RESULT="sequence":"1629759662963"
23:01:02.579 SL1: Send AT+SEND=ok

Behaviour of the LEDs don't change.  Under 8.2 there is no response to noise, on 9.5 it changes colour on noise no matter what.

**Reply by barbudor** [reply]
@blakadder @arendst It looks like the changes on Sonoff L1 introduced early this year for the MusicSync have some impact on standard Sonoff L1.
I can't see why
- ESP to MCU messages doesn't seem to have been impacted
- new L1MusicSync command doesn't seem to change that behavior
- Tasmota 9.2.0 is working on the same device while 9.3.0 doesn't

Could you please share your thought ?

**Reply by blakadder** [reply]
For me everyrhing works with latest Tasmota as it did before

**Reply by arendst** [reply]
For me too but as I said earlier in this thread I'll have a look where there could be differences. Just need time to do it (again).

**Reply by kyemacdonald** [reply]
Just for clarity it's a change with newer L1s. I have 2 L1 spiders. 1 new, 1 older (roughly 6 months). The older works with 9.5, the new one doesn't.

**Reply by blakadder** [reply]
how newer is it? mine is from an L1 lite with this PCB
![](https://blakadder.com/assets/images/l1lite/pcb_back.jpg)

**Reply by kyemacdonald** [reply]
![20210820_201518~2](https://user-images.githubusercontent.com/51263971/130578137-142fa7af-3450-4b12-96ca-c1676c7f1ec7.jpg)
It was ordered on the 9th of August. Arrived the day I posted this question.

Came as part of the L1 Lite smart led kit.

In terms of the pcb it looks identical.

**Reply by blakadder** [reply]
Same print on PCB should indicate no changes hardware wise

**Reply by barbudor** [reply]
> Same print on PCB should indicate no changes hardware wise

It could be the firmware. If protocol between ESP and MCU has changed ....
Apparently I see no difference in the code between 9.2.0 and 9.3+ regarding the structure of the AT+UPDATE message however it could be a timing thing...
That's why I was expecting that the original ESP firmware could have been saved before flashing Tasmota to allow some investigations.

**Comment by arendst** [comment]
I guess the quickest way to solve this is send me a Sonoff Lite to fiddle with it. Local supply seems to sell only the "old" version of the devices. iTead transport time is over a month.

**Reply by arendst** [reply]
Just ordered one from Amazon. Should be here by tomorrow but as it needs to come from Italy I doubt it. Anyway I'll let you know the status.

**Comment by arendst** [comment]
Just received my L1 lite from Italy. 

Time to make a copy of the code and start debugging!

**Comment by arendst** [comment, marked as answer]
Fixed in latest development release with commit https://github.com/arendst/Tasmota/commit/a6310780ee665394edb5c9b570acef4155f1b6a8

**Reply by mediarath** [reply]
Perfect. Works fine. Thx for the quick fix

**Reply by kyemacdonald** [reply]
Thanks Arendst,  confirmed working.

**Reply by martinsefcik** [reply]
Shouldn't be music sync mode be turned off automatically if I press for example any color on remote control ? Or do I have to explicitly always turn it off by `L1MusicSync 0` command?

**Reply by arendst** [reply]
Currently as you enabled musicsync with a command you'll have to turn it off that way too.

It's difficult to determine if a user pressed a button or that the microcontroller sends a status update. So turning it off that way opens up another can of worms where I suppose people are hardly using that musicsync mode anyway.

So for now keep using the command.

**Reply by martinsefcik** [reply]
OK, thanks for explanation.

**Reply by arendst** [reply]
Latest dev branch will turn off MusicSync when a valid remote control button is pressed. Makes very much sense indeed.
